src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -27,24 +27,25 @@ public List<Answer> getAnswers() {
         return Collections.unmodifiableList(answers);
     }
 
-    public void delete(NsUser writer) throws CannotDeleteException {
-        this.delete(writer, null);
-    }
-
-    public void delete(NsUser writer, DeleteHistories deleteHistories) throws CannotDeleteException {
+    private void delete(NsUser writer) throws CannotDeleteException {
         try {
             for (Answer answer : answers) {
                 answer.delete(writer);
-                if(deleteHistories != null) {
-                    deleteHistories.add(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now());
-                }
             }
         } catch (CannotDeleteException e) {
             throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
         }
     }
 
-    public boolean isDeleted(){
-        return answers.stream().allMatch(answer->answer.isDeleted());
+    public void delete(NsUser writer, DeleteHistories deleteHistories) throws CannotDeleteException {
+        this.delete(writer);
+
+        if (deleteHistories != null) {
+            deleteHistories.add(this);
+        }
```

Comment:
```suggestion
        if (deleteHistories != null) {
            answers.forEach(answer -> answer.addHistory(deleteHistories));
        }
```
- this를 전달하기 보다, 위와 같이 answer 객체에 메시지를 보내도 되지 않을까요?